src/main/java/com/bunq/sdk/context/UserContext.java

```diff
@@ -18,54 +19,41 @@
   private static final String ERROR_NO_ACTIVE_MONETARY_ACCOUNT_FOUND = "No active monetary account found.";
   private static final String ERROR_PRIMARY_MONETARY_ACCOUNT_IS_NOT_SET = "Primary monetaryAccount is not set.";
 
-  private static final String MONETARY_ACCOUNT_STATUS_ACTIVE = "ACTIVE";
-  private static final int INDEX_FIRST = 0;
-
+  private final ApiContext apiContext;
   private UserCompany userCompany;
   private UserPerson userPerson;
   private MonetaryAccountBank primaryMonetaryAccountBank;
-  private Integer userId;
-
-  public UserContext(Integer userId) {
-    this.setUser(this.getUserObject());
-    this.userId = userId;
-  }
 
-  private BunqModel getUserObject() {
-    return User.list().getValue().get(INDEX_FIRST).getReferencedObject();
+  public UserContext(ApiContext apiContext) {
+    this.apiContext = apiContext;
+    refreshContext();
   }
 
-  private void setUser(BunqModel user) {
-    if (user instanceof UserPerson) {
-      this.userPerson = (UserPerson) user;
-    } else if (user instanceof UserCompany) {
-      this.userCompany = (UserCompany) user;
+  private void initUser(User user) {
+    if (user!=null && user.getUserPerson()!=null) {
+      this.userPerson = user.getUserPerson();
+    } else if (user!=null && user.getUserCompany()!=null) {
+      this.userCompany = user.getUserCompany();
     } else {
       throw new BunqException(ERROR_UNEXPECTED_USER_INSTANCE);
     }
   }
 
-  public void initMainMonetaryAccount() {
-    List<MonetaryAccountBank> allMonetaryAccount =  MonetaryAccountBank.list().getValue();
-
-    for (MonetaryAccountBank monetaryAccountBank : allMonetaryAccount) {
-      if (monetaryAccountBank.getStatus().equals(MONETARY_ACCOUNT_STATUS_ACTIVE)) {
-        this.primaryMonetaryAccountBank = monetaryAccountBank;
-
-        return;
-      }
+  private void initMainMonetaryAccount(MonetaryAccountBank monetaryAccountBank) {
+    if(monetaryAccountBank==null) {
+      throw new BunqException(ERROR_NO_ACTIVE_MONETARY_ACCOUNT_FOUND);
```

Comment:
It does not make sense to throw this error here. It would made sense if this error says `monetaryAccountBank` can not be null. It being null will not automatically mean that no active MA has been found.

Reply:
the error was kept for consistency; the helper now returns first active monetaryAccountBank  or null, so if null it means no active monetaryAccountBank has been found